sycl/doc/extensions/GroupAlgorithms/SYCL_INTEL_group_sort.asciidoc

```diff
@@ -93,7 +191,9 @@ T operator()(Group g, T val);`
 Available only if `sycl::is_group_v<std::decay_t<Group>>` is true.
 |===
 
-Example of custom Sorter:
+Example of custom Sorter is below. This implementation is not recommended as a performant way
+to implement bubble sort. It's just an example of a custom sorter.

```

Comment:
Application can write a custom merge sorter with some specific optimizations, e.g. copy elements for disjoint sub-sequences. Users can also write the bitonic sorter that takes into account only sequences with 2^n elements. It can be faster than we can do since we need to work with any data sizes, but users can customize it for their specific cases. I'd suggest to have such example as a source file somewhere in "oneAPI Samples" repo to not overcomplicate the Spec.
Will it be better if we just remove the example of a custom sorter here, for the Spec?

Reply:
I assume that any of these example custom sorters would be significantly longer than the bubble sort we have now?  I agree that it would not be good to have a huge example in the spec.  If there is no short, reasonable example sorter, then maybe it is better not to have an example.

Regardless, I think it would be good to mention the use cases you list here.  Maybe something like:

> Custom sorters are particularly useful when the application knows the data has some special property.  For example, an application could implement a merge sort if it knows the data is already partially sorted.